apps/web/src/ee/billing/components/PlanHeader.tsx

```diff
@@ -47,28 +63,38 @@ export const PlanHeader = () => {
     any,
     any,
     { billingInterval: 'month' | 'year'; apiServiceLevel: ApiServiceLevelEnum }
-  >((data) => api.post('/v1/billing/checkout', data), {
-    onSuccess: (data) => {
-      if (upgradeOpen) {
-        return;
-      }
-
-      setIntentSecret(data.clientSecret);
-      setUpgradeOpen(true);
-    },
+  >((data) => api.post(checkoutUrl, data), {
+    onSuccess: checkoutOnSuccess,
     onError: (e: any) => {
       errorMessage(e.message || 'Unexpected error');
     },
   });
 
   useEffect(() => {
-    if (intentSecret === '') {
-      return;
+    if (!isStripeCheckoutEnabled) {
+      if (intentSecret === '') {
+        return;
+      }
+      checkout({ billingInterval, apiServiceLevel: ApiServiceLevelEnum.BUSINESS });
     }
-    checkout({ billingInterval, apiServiceLevel: ApiServiceLevelEnum.BUSINESS });
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [billingInterval, intentSecret, apiServiceLevel]);
 
+  useEffect(() => {
+    if (isStripeCheckoutEnabled) {
+      const query = new URLSearchParams(window.location.search);
+
+      if (query.get('success')) {
+        setApiServiceLevel(ApiServiceLevelEnum.BUSINESS);
+        successMessage('Payment was successful.');
+      }
+
+      if (query.get('canceled')) {
+        errorMessage('Order canceled.');
+      }
```

Comment:
Please handle the case for any other state other than success or canceled.

Reply:
Its only these two: https://docs.stripe.com/checkout/quickstart#urls

Reply:
Oh, I see.